### PR TITLE
feat: add neutral-focus recipe

### PR DESCRIPTION
**Pull request by nicholasrice**
# Description

Adds recipe for generating the neutral-focus color

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-styles-msft/src/utilities/color/neutral-focus.spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { neutralForegroundRest } from "./neutral-foreground";
+import { palette, Palette, PaletteType, Swatch } from "./palette";
+import designSystemDefaults, { DesignSystem } from "../../design-system";
+import { neutralFocus } from "./neutral-focus";
+
+describe("neutralForeground", (): void => {
```
Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/utilities/color/neutral-focus.spec.ts]: @nicholasrice assuming you mean "neutralFocus" here?

File: packages/fast-components-styles-msft/src/utilities/color/neutral-focus.spec.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    test("should return a string when invoked with an object", (): void => {
+        expect(typeof neutralFocus(designSystemDefaults)).toBe("string");
+    });
+
+    test("should return a function when invoked with a function", (): void => {
+        expect(typeof neutralFocus(() => "#FFF")).toBe("function");
+    });
+
+    test("should operate on default design system if no design system is supplied", (): void => {
+        expect(neutralFocus({} as DesignSystem)).toBe("#101010");
```
Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/utilities/color/neutral-focus.spec.ts]: Do we want this test to fail if we change the default design system value? Should we expect the empty object to match the same function with defaults passed?

Comment (nicholasrice) [reply packages/fast-components-styles-msft/src/utilities/color/neutral-focus.spec.ts]: I think we want this to fail if we change the default design-system to something that makes this change, yea -IMO that would be a very significant change that we would want to know about if done inadvertently.

**Comment by nicholasrice** [comment]
closes #1461

**Pull request merged**